lib/react_on_rails/react_component/render_options.rb

```diff
@@ -111,8 +111,32 @@ def set_option(key, value)
         options[key] = value
       end
 
-      def stream?
-        options[:stream?]
+      def render_mode
+        # Determines the React rendering strategy:
+        # - :sync: Synchronous SSR using renderToString (blocking and rendering in one shot)
+        # - :html_streaming: Progressive SSR using renderToPipeableStream (non-blocking and rendering incrementally)
+        # - :rsc_payload_streaming: Server Components serialized in React flight format
+        #   (non-blocking and rendering incrementally).
+        options.fetch(:render_mode, :sync)
+      end
+
+      def streaming?
+        # Returns true if the component should be rendered incrementally
+        %i[html_streaming rsc_payload_streaming].include?(render_mode)
+      end
+
+      def rsc_payload_streaming?
+        # Returns true if the component should be rendered as a React Server Component
+        render_mode == :rsc_payload_streaming
+      end
+
+      def html_streaming?
+        # Returns true if the component should be rendered incrementally
+        render_mode == :html_streaming
+      end
+
+      def store_dependencies
+        options[:store_dependencies]
```

Comment:
Could we, should we, have some specialized code for RSC be in the React on Rails Pro source?

Reply:
We do in https://github.com/shakacode/react_on_rails_pro/pull/422, which you should also review.